### ISSUE TITLE
Ignoring erros on iOS 14

### ISSUE THREAD

**Issue by santa010** [flutterchina/dio#952]
### New Issue Checklist

- [ X ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |  flutter   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |  Channel beta, 1.22.0-12.1.pre    |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   3.0.10   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |   Android Studio 4.0.1   |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |   100%   |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Can't catch any exceptions on iOS 14 (working normally on android and iOS 10).
Example:
```
    try {
      // Try some invalid address
      Response response = await Dio().get("http://wwwx.google.com");
      print(response);
    } catch (e) {
      print(e);
    }
``` 
There is also no DioError when firing a request without internet connection.

**Comment by JagandeepBrar** [comment]
Experiencing the same issue, no matter what I try I cannot get any errors to be caught anymore.

Tried wrapping the call in a try-catch block as OP did, tried `.catchError` directly on the Dio call, also doesn't work with chained futures.
